### PR TITLE
Closes #15237: Add tests for missing filters

### LINKED ISSUE #15237: Audit FilterSet definitions application-wide
### Proposed Changes

 Review all currently defined FilterSet classes and check for:

- Missing filters
- Inconsistent names or behavior
- Suboptimal performance
- Opportunities for code reduction

### Justification

The FilterSet classes defined in NetBox have evolved organically over the past several years and we are overdue for a structured audit of the existing filter sets. This will help us identify inconsistent or suboptimal implementations, and may yield additional work to help standardize and shrink the code base.

### PR DESCRIPTION
**Pull request by jeremystretch**
### Fixes: #15237

- Introduces the `test_missing_filters()` method on `BaseFilterSetTests`, which check that all relevant model fields have the appropriate filter(s) defined on the model's filter set.
- Implements the missing filters identified by the new tests
- Add functional tests for the new filters

### CODE REVIEW

File: netbox/dcim/filtersets.py [hunk 1]
```diff
@@ -447,10 +447,14 @@ class RackReservationFilterSet(NetBoxModelFilterSet, TenancyFilterSet):
         to_field_name='username',
         label=_('User (name)'),
     )
+    unit = NumericArrayFilter(
+        field_name='units',
+        lookup_expr='contains'
+    )
```
Comment (arthanson) [inline netbox/dcim/filtersets.py:453]: label?  Not completely sure these need labels, I guess the REST documentation would show it.  Noticed there are a lot missing in the filtersets.

File: netbox/dcim/filtersets.py [hunk 2]
```diff
@@ -751,18 +777,21 @@ class InterfaceTemplateFilterSet(ChangeLoggedModelFilterSet, ModularDeviceTypeCo
...
 
 
 class FrontPortTemplateFilterSet(ChangeLoggedModelFilterSet, ModularDeviceTypeComponentFilterSet):
     type = django_filters.MultipleChoiceFilter(
         choices=PortTypeChoices,
         null_value=None
     )
+    rear_port_id = django_filters.ModelMultipleChoiceFilter(
+        queryset=RearPort.objects.all()
+    )
```
Comment (arthanson) [inline netbox/dcim/filtersets.py:790]: label?  Noticed FrontPortTemplateFilterSet above is also missing label on the fields.

File: netbox/dcim/filtersets.py [hunk 3]
```diff
@@ -1583,13 +1646,23 @@ class InterfaceFilterSet(
         to_field_name='name',
         label='Virtual Device Context',
     )
+    wireless_lan_id = django_filters.ModelMultipleChoiceFilter(
+        field_name='wireless_lans',
+        queryset=WirelessLAN.objects.all(),
+        label='Wireless LAN',
```
Comment (arthanson) [inline netbox/dcim/filtersets.py:1652]: internationalize

File: netbox/dcim/filtersets.py [hunk 4]
```diff
@@ -1583,13 +1646,23 @@ class InterfaceFilterSet(
...
+    )
+    wireless_link_id = django_filters.ModelMultipleChoiceFilter(
+        queryset=WirelessLink.objects.all(),
+        label='Wireless link',
+    )
```
Comment (arthanson) [inline netbox/dcim/filtersets.py:1657]: internationalize

File: netbox/dcim/filtersets.py [hunk 5]
```diff
@@ -1618,10 +1691,15 @@ class FrontPortFilterSet(
         choices=PortTypeChoices,
         null_value=None
     )
+    rear_port_id = django_filters.ModelMultipleChoiceFilter(
+        queryset=RearPort.objects.all()
+    )
```
Comment (arthanson) [inline netbox/dcim/filtersets.py:1696]: label

File: netbox/extras/filtersets.py [hunk 6]
```diff
@@ -220,13 +227,14 @@ def search(self, queryset, name, value):
...
+class ExportTemplateFilterSet(ChangeLoggedModelFilterSet):
     q = django_filters.CharFilter(
         method='search',
         label=_('Search'),
     )
-    object_type_id = MultiValueNumberFilter(
-        field_name='object_types__id'
+    object_type_id = django_filters.ModelMultipleChoiceFilter(
+        queryset=ObjectType.objects.all(),
+        field_name='object_types'
```
Comment (arthanson) [inline netbox/extras/filtersets.py:237]: label?

File: netbox/extras/filtersets.py [hunk 7]
```diff
@@ -253,13 +264,14 @@ def search(self, queryset, name, value):
...
+class SavedFilterFilterSet(ChangeLoggedModelFilterSet):
     q = django_filters.CharFilter(
         method='search',
         label=_('Search'),
     )
-    object_type_id = MultiValueNumberFilter(
-        field_name='object_types__id'
+    object_type_id = django_filters.ModelMultipleChoiceFilter(
+        queryset=ObjectType.objects.all(),
+        field_name='object_types'
```
Comment (arthanson) [inline netbox/extras/filtersets.py:274]: label?

File: netbox/ipam/filtersets.py [hunk 8]
```diff
@@ -882,8 +930,9 @@ def search(self, queryset, name, value):
         return queryset.filter(qs_filter)
 
     def filter_scope(self, queryset, name, value):
+        model_name = name.replace('_', '')
         return queryset.filter(
-            scope_type=ContentType.objects.get(model=name),
+            scope_type=ContentType.objects.get(model=model_name),
```
Comment (arthanson) [inline netbox/ipam/filtersets.py:935]: I assume this is a separate bug-fix?

File: netbox/users/filtersets.py [hunk 9]
```diff
@@ -115,6 +135,13 @@ class ObjectPermissionFilterSet(BaseFilterSet):
         method='search',
         label=_('Search'),
     )
+    object_type_id = django_filters.ModelMultipleChoiceFilter(
+        queryset=ObjectType.objects.all(),
+        field_name='object_types'
```
Comment (arthanson) [inline netbox/users/filtersets.py:140]: label?

File: netbox/vpn/filtersets.py [hunk 10]
```diff
@@ -158,13 +169,20 @@ class IKEPolicyFilterSet(NetBoxModelFilterSet):
...
+    ike_proposal_id = django_filters.ModelMultipleChoiceFilter(
+        field_name='proposals',
+        queryset=IKEProposal.objects.all()
     )
-    proposal = MultiValueCharFilter(
-        field_name='proposals__name'
+    ike_proposal = django_filters.ModelMultipleChoiceFilter(
+        field_name='proposals__name',
+        queryset=IKEProposal.objects.all(),
+        to_field_name='name'
```
Comment (arthanson) [inline netbox/vpn/filtersets.py:179]: label?

File: netbox/vpn/filtersets.py [hunk 11]
```diff
@@ -205,13 +234,20 @@ class IPSecPolicyFilterSet(NetBoxModelFilterSet):
     pfs_group = django_filters.MultipleChoiceFilter(
         choices=DHGroupChoices
     )
-    proposal_id = MultiValueNumberFilter(
-        field_name='proposals__id'
+    ipsec_proposal_id = django_filters.ModelMultipleChoiceFilter(
+        field_name='proposals',
+        queryset=IPSecProposal.objects.all()
```
Comment (arthanson) [inline netbox/vpn/filtersets.py:239]: label?

File: netbox/wireless/filtersets.py [hunk 12]
```diff
@@ -60,6 +61,10 @@ class WirelessLANFilterSet(NetBoxModelFilterSet, TenancyFilterSet):
     vlan_id = django_filters.ModelMultipleChoiceFilter(
         queryset=VLAN.objects.all()
     )
+    interface_id = django_filters.ModelMultipleChoiceFilter(
+        queryset=Interface.objects.all(),
+        field_name='interfaces'
```
Comment (arthanson) [inline netbox/wireless/filtersets.py:66]: label?

**Review by arthanson — CHANGES_REQUESTED** [review]
Looks very good, mostly internationalization (which could be a separate PR as it looks like a sweep of filtersets is needed).  Also standardization on tuples for fields (again could be a separate PR)

File: netbox/ipam/filtersets.py (hunk 8 above)
Comment (jeremystretch) [reply netbox/ipam/filtersets.py:935]: This change was just to accommodate renaming the two filters above (`sitegroup` and `clustergroup`).

File: netbox/dcim/filtersets.py (hunk 3 above)
Comment (jeremystretch) [reply netbox/dcim/filtersets.py:1652]: Found a few instances where we neglected to wrap labels with `gettext()`; going to clean them all up.

**Comment by jeremystretch** [comment]
Thanks @arthanson. I'm going to punt on the missing labels for now; we should spend some time figuring out where labels are actually needed (will address this in a separate future issue).

**Pull request merged**